### ISSUE TITLE
Mangled function signature when using lru_cache

### ISSUE THREAD

**Issue by wtbarnes** [wtbarnes/fiasco#73]
#69 wrapped `fiasco.list_ions` with the `functools.lru_cache` decorator to greatly increase its performance. This has the unfortunate side affect of messing up the autogenerated API docs. For some reason, automod-api cannot sort out the current module or correction function name.

Is there something we can do about this? maybe wrapping the decorator? [This blog post](https://hynek.me/articles/decorators/) has some interesting related information. Should this be reported upstream to the sphinx-automodapi extension?

**Comment by dstansby** [comment]
I would say maybe report upstream? The blog post seems to say that using `inspect.signature()` to get the signature solves the issue, so maybe this can be incorporated into sphinx-automodapi somehow?

**Comment by dstansby** [comment]
Possibly related: https://github.com/sphinx-doc/sphinx/issues/6588

**Comment by dstansby** [comment]
A bit more investigation: `automodapi` generates the following file:

```rst
fiasco.list_ions
================

.. currentmodule::

.. autoattribute:: fiasco.list_ions
```
using this template: https://github.com/astropy/sphinx-automodapi/blob/master/sphinx_automodapi/templates/autosummary_core/base.rst

So, I think the problem is that `sphinx.ext.autosummary` is getting the `objname` and `module` variables wrong and the problem lies there.

**Comment by dstansby** [comment]
More digging: this seems to boil down to `inspect.isfunction(fiasco.list_ions)` returning `False` when it is decorated...

... because
```python
>>> type(fiasco.list_ions)
<class 'functools._lru_cache_wrapper'>
```

**Comment by wtbarnes** [comment]
Closed by #81
